File: scripts/plot_angles.R
# alternative R plotting script (need ggplot2, dplyr, readr)
library(ggplot2)
library(dplyr)
library(readr)

df <- read_tsv("final/angles.tsv", show_col_types = FALSE)

df$size_class <- factor(df$size_class,
                        levels = c("Tiny", "Small", "Intermediate", "Large", "Bulky"))

df <- df %>% mutate(angle = ((angle + 180) %% 360) - 180)

pal <- c(
  "Tiny"         = "#c8c4b5",
  "Small"        = "#e8b44c",
  "Intermediate" = "#e07840",
  "Large"        = "#c93d1d",
  "Bulky"        = "#8b0a1a"
)

n <- nrow(df)

p <- ggplot(df, aes(x = angle, color = size_class)) +
  geom_density(linewidth = 1.2, adjust = 1.2) +
  scale_color_manual(values = pal) +
  scale_x_continuous(limits = c(-180, 180), breaks = seq(-150, 150, 50)) +
  labs(
    title = paste0("Tripeptide (XRX) in Helix (n = ", n, ")"),
    x = "Angle between adjacent CA-Centroid vectors (degrees)",
    y = "Norm. Freq. [A.U.]",
    color = NULL
  ) +
  theme_minimal(base_size = 14) +
  theme(
    panel.background  = element_rect(fill = "#bdbdbd", color = NA),
    plot.background   = element_rect(fill = "#bdbdbd", color = NA),
    panel.grid.major.x = element_line(color = "blue", linetype = "dotted"),
    panel.grid.major.y = element_blank(),
    panel.grid.minor   = element_blank(),
    legend.position    = c(0.15, 0.85),
    legend.background  = element_rect(fill = "white", color = "black"),
    plot.title         = element_text(hjust = 0.5)
  )

ggsave("final/angle_plot.png", plot = p, width = 10, height = 6, dpi = 300)

File: scripts/plot_angles.py
# plot angle distributions grouped by residue size class
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import gaussian_kde

df = pd.read_csv("final/angles.tsv", sep="\t")

# make sure angles are in [-180, 180]
df["angle"] = ((df["angle"] + 180) % 360) - 180

size_order = ["Tiny", "Small", "Intermediate", "Large", "Bulky"]
colors = {
    "Tiny":         "#c8c4b5",
    "Small":        "#e8b44c",
    "Intermediate": "#e07840",
    "Large":        "#c93d1d",
    "Bulky":        "#8b0a1a",
}

fig, ax = plt.subplots(figsize=(10, 6))
fig.patch.set_facecolor("#bdbdbd")
ax.set_facecolor("#bdbdbd")

xs = np.linspace(-180, 180, 500)

for sz in size_order:
    vals = df[df["size_class"] == sz]["angle"].values
    if len(vals) < 2:
        continue
    kde = gaussian_kde(vals, bw_method=0.15)
    ax.plot(xs, kde(xs), color=colors[sz], linewidth=1.5, label=sz)

# vertical grid lines
for x in range(-150, 200, 50):
    ax.axvline(x, color="blue", linestyle=":", linewidth=0.5, alpha=0.6)

ax.set_xlim(-180, 180)
ax.set_xticks(range(-150, 200, 50))
ax.set_xlabel(u"Angle between adjacent C-\u03b1 -> Centroid vectors [\u00b0]", fontsize=13)
ax.set_ylabel("Norm. Freq. [A.U.]", fontsize=13)
ax.set_title(f"Tripeptide (XRX) in Helix (n = {len(df)})", fontsize=14)
ax.legend(loc="upper left", frameon=True, facecolor="white", edgecolor="black")
ax.tick_params(axis="y", left=False, labelleft=True)
ax.grid(False)

plt.tight_layout()
plt.savefig("final/angle_plot.png", dpi=300, facecolor="#bdbdbd")
print(f"saved plot to final/angle_plot.png (n={len(df)})")
